src/env-inl.h

```diff
@@ -172,7 +172,8 @@ inline bool TickInfo::has_rejection_to_warn() const {
 }
 
 inline Environment* Environment::GetCurrent(v8::Isolate* isolate) {
-  if (UNLIKELY(!isolate->InContext())) return nullptr;
+  if (UNLIKELY(!isolate->InContext() ||
+               isolate->IsExecutionTerminating())) return nullptr;
```

Comment:
Can you leave a comment here about why this is necessary? I’d definitely like to avoid a change like this, esp. considering that this method should never attempt to call into JS anyway, so if it’s about signaling something specific to callers then it’s probably worth thinking about whether there’s another way of doing this. (If it’s about the `HandleScope` then I’d suggest to conditionally omit that scope, but that seems like it would cause trouble for us in other places as well.)

Reply:
Yes, it was because of the `HandleScope` and I agree this is not the place to make this change. I'm trying another approach in the next patch.